### ISSUE TITLE
(scalar localtime)[0,0] gives ('...',undef)

### ISSUE THREAD

**Issue by p5pRT** [Perl/perl5#4541]
Migrated from [rt.perl.org#7862](https://rt-archive.perl.org/perl5/Ticket/Display.html?id=7862) (status was 'resolved')

Searchable as RT7862$

**Comment by p5pRT** [comment]
### From tye@metronet.com


\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-\-
\(scalar localtime\)\[0\,0\] gives \('\.\.\.'\,undef\)

Several versions of Perl exhibit the behavior that

&nbsp;    \(scalar localtime\)\[0\,0\]

yields \('Thu Nov  1 12&#8203;:04&#8203;:40 2001'\,undef\) instead of two
copies of the same string\.  Similar results are found when
replacing localtime\(\) with gmtime\(\) or each\(\) but not for
most other expressions\.


Here is "perl \-V" for a second version of Perl that exhibits this
problem&#8203;:

Summary of my perl5 \(5\.0 patchlevel 4 subversion 1\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=hpux\, osvers=9\, archname=PA\-RISC1\.1
&nbsp;    uname='hp\-ux fiinix a\.09\.05 a 9000705 2004910115 two\-user license '
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    bincompat3=n useperlio= d\_sfio=
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, optimize='\-O'\, gccversion=2\.6\.3
&nbsp;    cppflags='\-D\_POSIX\_SOURCE \-DDEBUGGING\_MSTATS \-DEMERGENCY\_SBRK \-DPACK\_MALLOC \-DTWO\_POT\_OPTIMIZE \-D\_HPUX\_SOURCE \-L/lib/pa1\.1'
&nbsp;    ccflags ='\-D\_POSIX\_SOURCE \-DDEBUGGING\_MSTATS \-DEMERGENCY\_SBRK \-DPACK\_MALLOC \-DTWO\_POT\_OPTIMIZE \-D\_HPUX\_SOURCE \-L/lib/pa1\.1'
&nbsp;    stdchar='unsigned char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    voidflags=15\, castflags=0\, d\_casti32=define\, d\_castneg=define
&nbsp;    intsize=4\, alignbytes=8\, usemymalloc=y\, randbits=15
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='ld'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib/pa1\.1 /lib /usr/lib
&nbsp;    libs=\-lndbm \-lgdbm \-ldb \-ldld \-lm \-lc
&nbsp;    libc=/lib/libc\.sl\, so=sl
&nbsp;    useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_hpux\.xs\, dlext=sl\, d\_dlsymun=\, ccdlflags='\-Wl\,\-E \-Wl\,\-B\,deferred '
&nbsp;    cccdlflags='\-fpic'\, lddlflags='\-b \-L/usr/local/lib'


Characteristics of this binary \(from libperl\)&#8203;: 
&nbsp;  Built under hpux
&nbsp;  Compiled at Jun 30 1997 23&#8203;:36&#8203;:57
&nbsp;  %ENV&#8203;:
&nbsp;    PERLLIB="/s/t/tye/perllib"
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /s/t/tye/perllib
&nbsp;    /usr/local/lib/perl5\.004/lib/PA\-RISC1\.1/5\.00401
&nbsp;    /usr/local/lib/perl5\.004/lib
&nbsp;    /usr/local/lib/perl5\.004/metronet/PA\-RISC1\.1
&nbsp;    /usr/local/lib/perl5\.004/metronet
&nbsp;    \.

<details><summary>Perl Info</summary>

```

Flags:
    category=core
    severity=medium

Site configuration information for perl v5.6.0:

Configured by etm at Fri Jul 14 13:46:57 2000.

Summary of my perl5 (revision 5 version 6 subversion 0) configuration:
  Platform:
    osname=MSWin32, osvers=4.0, archname=MSWin32-x86-multi-thread
    uname=''
    config_args='undef'
    hint=recommended, useposix=true, d_sigaction=undef
    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=define
    useperlio=undef d_sfio=undef uselargefiles=undef 
    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
  Compiler:
    cc='cl', optimize='-O1 -MD -DNDEBUG', gccversion=
    cppflags='-DWIN32'
    ccflags ='-O1 -MD -DNDEBUG -DWIN32 -D_CONSOLE -DNO_STRICT -DHAVE_DES_FCRYPT  -DPERL_IMPLICIT_CONTEXT -DPERL_IMPLICIT_SYS -DPERL_MSVCRT_READFIX'
    stdchar='char', d_stdstdio=define, usevfork=false
    intsize=4, longsize=4, ptrsize=4, doublesize=8
    d_longlong=undef, longlongsize=8, d_longdbl=define, longdblsize=10
    ivtype='long', ivsize=4, nvtype='double', nvsize=8, Off_t='off_t', lseeksize=4
    alignbytes=8, usemymalloc=n, prototype=define
  Linker and Libraries:
    ld='link', ldflags ='-nologo -nodefaultlib -release  -libpath:"F:\Apps\Perl\lib\CORE"  -machine:x86'
    libpth="F:\Apps\Perl\lib\CORE"
    libs=  oldnames.lib kernel32.lib user32.lib gdi32.lib winspool.lib  comdlg32.lib advapi32.lib shell32.lib ole32.lib oleaut32.lib  netapi32.lib uuid.lib wsock32.lib mpr.lib winmm.lib  version.lib odbc32.lib odbccp32.lib msvcrt.lib
    libc=msvcrt.lib, so=dll, useshrplib=yes, libperl=perl56.lib
  Dynamic Linking:
    dlsrc=dl_win32.xs, dlext=dll, d_dlsymun=undef, ccdlflags=' '
    cccdlflags=' ', lddlflags='-dll -nologo -nodefaultlib -release  -libpath:"F:\Apps\Perl\lib\CORE"  -machine:x86'

Locally applied patches:
    ACTIVEPERL_LOCAL_PATCHES_ENTRY


@INC for perl v5.6.0:
    F:/Apps/Perl/lib
    F:/Apps/Perl/site/lib
    .


Environment for perl v5.6.0:
    HOME (unset)
    LANG (unset)
    LANGUAGE (unset)
    LD_LIBRARY_PATH (unset)
    LOGDIR (unset)
    PATH=F:\Apps\Perl\bin;F:\Apps\iPerl\bin;F:\Apps\DosBin;C:\WINNT\system32;C:\WINNT;F:\Apps\Cygwin\bin;F:\Apps\Cygwin\contrib\bin;f:\apps\Sql7\BINN
    PERL_BADLANG (unset)
    SHELL (unset)

```
</details>

**Comment by p5pRT** [comment]
### From @nwc10

Odd&#8203;:

$ perl \-lwe 'print \+ \(scalar localtime\)\[0\,0\]'
Thu Nov  1 20&#8203;:15&#8203;:40 2001Thu Nov  1 20&#8203;:15&#8203;:40 2001
$ perl \-lwe 'print \+ @&#8203;a = \(scalar localtime\)\[0\,0\]'
Use of uninitialized value at \-e line 1\.
Thu Nov  1 20&#8203;:15&#8203;:53 2001

\(that's 5\.004\_04\)

all the 5\.7\.x I have here don't show it\,

$ perl5\.6\.1 \-lwe 'print \+ @&#8203;a = \(scalar localtime\)\[0\,0\]'
Name "main&#8203;::a" used only once&#8203;: possible typo at \-e line 1\.
Use of uninitialized value in print at \-e line 1\.
Thu Nov  1 20&#8203;:17&#8203;:39 2001
$ perl5\.6\.1 \-lwe 'print \+ \(scalar localtime\)\[0\,0\]'
Thu Nov  1 20&#8203;:17&#8203;:46 2001Thu Nov  1 20&#8203;:17&#8203;:46 2001

\[this is usually the point where someone else steps in with the patch that
fixed it\, and I look a bit silly\]

Nicholas Clark

**Comment by p5pRT** [comment]
### From [Unknown Contact. See original ticket]

> This is a bug report for perl from tye@&#8203;metronet\.com\,
> generated with the help of perlbug 1\.28 running under perl v5\.6\.0\.

Other versions were found by the PerlMonks to have the same behavior
with the source

&nbsp;  \($aa\, $bb\) = \(scalar localtime\)\[0\,0\];

as what Tye reports for his versions\. The other versions include
5\.6\.0 \(ActiveState 623\) for Windows\, 5\.6\.0 for Linux\, and 5\.005\_03
for Linux\. Here are my perl \-V results&#8203;:

Summary of my perl5 \(revision 5 version 6 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=MSWin32\, osvers=4\.0\, archname=MSWin32\-x86\-multi\-thread
&nbsp;    uname=''
&nbsp;    config\_args='undef'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=undef
&nbsp;    usethreads=undef use5005threads=undef useithreads=define usemultiplicity=def
ine
&nbsp;    useperlio=undef d\_sfio=undef uselargefiles=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cl'\, optimize='\-O1 \-MD \-DNDEBUG'\, gccversion=
&nbsp;    cppflags='\-DWIN32'
&nbsp;    ccflags ='\-O1 \-MD \-DNDEBUG \-DWIN32 \-D\_CONSOLE \-DNO\_STRICT \-DHAVE\_DES\_FCRYPT
 \-DPERL\_IMPLICIT\_CONTEXT \-DPERL\_IMPLICIT\_SYS \-DPERL\_MSVCRT\_READFIX'
&nbsp;    stdchar='char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=undef\, longlongsize=8\, d\_longdbl=define\, longdblsize=10
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize
=4
&nbsp;    alignbytes=8\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='link'\, ldflags ='\-nologo \-nodefaultlib \-release  \-libpath&#8203;:"C&#8203;:\\Perl\\lib\\C
ORE"  \-machine&#8203;:x86'
&nbsp;    libpth="C&#8203;:\\Perl\\lib\\CORE"
&nbsp;    libs=  oldnames\.lib kernel32\.lib user32\.lib gdi32\.lib winspool\.lib  comdlg32
\.\.lib advapi32\.lib shell32\.lib ole32\.lib oleaut32\.lib  netapi32\.lib uuid\.lib wsoc
k32\.lib mpr\.lib winmm\.lib  version\.lib odbc32\.lib odbccp32\.lib msvcrt\.lib
&nbsp;    libc=msvcrt\.lib\, so=dll\, useshrplib=yes\, libperl=perl56\.lib
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_win32\.xs\, dlext=dll\, d\_dlsymun=undef\, ccdlflags=' '
&nbsp;    cccdlflags=' '\, lddlflags='\-dll \-nologo \-nodefaultlib \-release  \-libpath&#8203;:"C&#8203;:
\\Perl\\lib\\CORE"  \-machine&#8203;:x86'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;: MULTIPLICITY USE\_ITHREADS PERL\_IMPLICIT\_CONTEXT PERL\_IMP
LICIT\_SYS
&nbsp;  Locally applied patches&#8203;:
&nbsp;        ActivePerl Build 623
&nbsp;  Built under MSWin32
&nbsp;  Compiled at Dec 15 2000 16&#8203;:27&#8203;:07
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    C&#8203;:/Perl/lib
&nbsp;    C&#8203;:/Perl/site/lib
&nbsp;    \.

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#3

Summary of my perl5 \(revision 5\.0 version 6 subversion 0\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.2\.17\-8smp\, archname=i386\-linux
&nbsp;    uname='linux porky\.devel\.redhat\.com 2\.2\.17\-8smp \#1 smp fri nov 17 16&#8203;:12&#8203;:17 est 2000 i686 unknown '
&nbsp;    config\_args='\-des \-Doptimize=\-O2 \-march=i386 \-mcpu=i686 \-Dcc=gcc \-Dcccdlflags=\-fPIC \-Dinstallprefix=/usr \-Dprefix=/usr \-Darchname=i386\-linux \-Dd\_dosuid \-Dd\_semctl\_semun \-Di\_db \-Di\_ndbm \-Di\_gdbm \-Di\_shadow \-Di\_syslog \-Dman3ext=3pm \-Uuselargefiles'
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef use5005threads=undef useithreads=undef usemultiplicity=undef
&nbsp;    useperlio=undef d\_sfio=undef uselargefiles=undef
&nbsp;    use64bitint=undef use64bitall=undef uselongdouble=undef usesocks=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='gcc'\, optimize='\-O2 \-march=i386 \-mcpu=i686'\, gccversion=2\.96 20000731 \(Red Hat Linux 7\.1 2\.96\-79\)
&nbsp;    cppflags='\-fno\-strict\-aliasing'
&nbsp;    ccflags ='\-fno\-strict\-aliasing'
&nbsp;    stdchar='char'\, d\_stdstdio=define\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    ivtype='long'\, ivsize=4\, nvtype='double'\, nvsize=8\, Off\_t='off\_t'\, lseeksize=4
&nbsp;    alignbytes=4\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='gcc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-ldl \-lm \-lc \-lcrypt
&nbsp;    libc=/lib/libc\-2\.2\.2\.so\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic'
&nbsp;    cccdlflags='\-fPIC'\, lddlflags='\-shared \-L/usr/local/lib'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Compile\-time options&#8203;:
&nbsp;  Built under linux
&nbsp;  Compiled at Mar 23 2001 12&#8203;:49&#8203;:50
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /usr/lib/perl5/5\.6\.0/i386\-linux
&nbsp;    /usr/lib/perl5/5\.6\.0
&nbsp;    /usr/lib/perl5/site\_perl/5\.6\.0/i386\-linux
&nbsp;    /usr/lib/perl5/site\_perl/5\.6\.0
&nbsp;    /usr/lib/perl5/site\_perl
&nbsp;    \.

\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#\#

Summary of my perl5 \(5\.0 patchlevel 5 subversion 3\) configuration&#8203;:
&nbsp;  Platform&#8203;:
&nbsp;    osname=linux\, osvers=2\.2\.5\-22smp\, archname=i386\-linux
&nbsp;    uname='linux porky\.devel\.redhat\.com 2\.2\.5\-22smp \#1 smp wed jun 2 09&#8203;:11&#8203;:51 edt 1999 i686 unknown '
&nbsp;    hint=recommended\, useposix=true\, d\_sigaction=define
&nbsp;    usethreads=undef useperlio=undef d\_sfio=undef
&nbsp;  Compiler&#8203;:
&nbsp;    cc='cc'\, optimize='\-O2 \-m486 \-fno\-strength\-reduce'\, gccversion=egcs\-2\.91\.66 19990314/Linux \(egcs\-1\.1\.2 release\)
&nbsp;    cppflags='\-Dbool=char \-DHAS\_BOOL \-I/usr/local/include'
&nbsp;    ccflags ='\-Dbool=char \-DHAS\_BOOL \-I/usr/local/include'
&nbsp;    stdchar='char'\, d\_stdstdio=undef\, usevfork=false
&nbsp;    intsize=4\, longsize=4\, ptrsize=4\, doublesize=8
&nbsp;    d\_longlong=define\, longlongsize=8\, d\_longdbl=define\, longdblsize=12
&nbsp;    alignbytes=4\, usemymalloc=n\, prototype=define
&nbsp;  Linker and Libraries&#8203;:
&nbsp;    ld='cc'\, ldflags =' \-L/usr/local/lib'
&nbsp;    libpth=/usr/local/lib /lib /usr/lib
&nbsp;    libs=\-lnsl \-ldl \-lm \-lc \-lposix \-lcrypt
&nbsp;    libc=\, so=so\, useshrplib=false\, libperl=libperl\.a
&nbsp;  Dynamic Linking&#8203;:
&nbsp;    dlsrc=dl\_dlopen\.xs\, dlext=so\, d\_dlsymun=undef\, ccdlflags='\-rdynamic'
&nbsp;    cccdlflags='\-fpic'\, lddlflags='\-shared \-L/usr/local/lib'


Characteristics of this binary \(from libperl\)&#8203;:
&nbsp;  Built under linux
&nbsp;  Compiled at Aug 10 2000 15&#8203;:33&#8203;:00
&nbsp;  @&#8203;INC&#8203;:
&nbsp;    /usr/lib/perl5/5\.00503/i386\-linux
&nbsp;    /usr/lib/perl5/5\.00503
&nbsp;    /usr/lib/perl5/site\_perl/5\.005/i386\-linux
&nbsp;    /usr/lib/perl5/site\_perl/5\.005
&nbsp;    \.

 
\-\- 
Programming is a tool\. A tool is neither good nor evil\. It is
the user who determines how it is used and to what ends\.

\-\-\-\-\- End forwarded message \-\-\-\-\-

\-\- 
If they can get you asking the wrong questions\, they don't
have to worry about the answers\.
&nbsp;  \-\- Thomas Pynchon\, Gravity's Rainbow

**Comment by p5pRT** [comment]
### From @andk

&nbsp;  > \[this is usually the point where someone else steps in with the patch that
&nbsp;  > fixed it\, and I look a bit silly\]

Yeah\, Nicholas\, off the top of my head I'd say that one was 7867 :\-\)

\-\-> Seems like a duplicate of 20000212\.002

\-\- 
andreas

**Comment by p5pRT** [comment]
@iabyn - Status changed from 'open' to 'resolved'
